src/main/java/com/customerApi/api/controllers/CustomerController.java

```diff
@@ -0,0 +1,86 @@
+package com.customerApi.api.controllers;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.customerApi.api.modals.Customer;
+import com.customerApi.api.services.CustomerService;
+
+@RestController
+public class CustomerController {
+	private static final Logger logger = LoggerFactory.getLogger(CustomerController.class);
+
+	@Autowired
+	private CustomerService service;
+
+	@PostMapping("/save")
+	public ResponseEntity<Customer> saveCustomer(@Valid @RequestBody Customer data) {
+
+		service.addCustomerData(data);
+
+		return new ResponseEntity<>(data, HttpStatus.OK);
+
+	}
+
+	@GetMapping("/getCustomer")
+	@ResponseBody
+	public List<Customer> getCustomer() {
+
+		List<Customer> data = service.getCustomersData();
+
+		return data;
+	}
+
+	@GetMapping("getCustomer/{id}")
+	@ResponseBody
+	public Optional<Customer> getCustomerById(@Valid @PathVariable("id") int id) {
```

Comment:
1. Response should be ResponseEntity, I told you last time too you complete code should be in Sync, you should follow the same standards around all the endpoints. So fix it and return ResponseEntity in response.
2. Are you sure this @**Valid** will work here if you don't pass **id** in request?